nextroute/model_expression_custom.go

```diff
@@ -136,6 +155,19 @@ type VehicleTypeExpression interface {
 	)
 }
 
+// VehicleTypeDistanceExpression is a distance expression that has a value for
+// each vehicle type.
+type VehicleTypeDistanceExpression interface {
+	VehicleExpression
+
+	// SetValue sets the distance value of the expression for the given
```

Comment:
This can stay the same if we decide by making VehicleTypeDistanceExpression a DefaultExpression implementation and otherwise you have to change it to SetDistance (no method overloading allowed)